### PR TITLE
[18.0][ADD] website_blog_scheduled_publication: add new module

### PR DESCRIPTION
**Pull request by CristianoMafraJunior**
## Summary SuperSeed #1159
This PR adds a new module website_blog_scheduled_publication that allows users to schedule blog posts for automatic publication at a specific date and time.

### Features
- **Scheduled Publication Date:** Adds a datetime field to blog posts for scheduling publication
- **Quick Publish Button:** One-click immediate publication from the form view button box
- **Schedule Wizard:** User-friendly wizard dialog to set the publication date
- **Visual Feedback:** Information banner displays the scheduled date on the post form
- **Smart Publication Logic:** Automatically prevents immediate publication if a future date is set
- **Automatic Cron Job:** Runs every hour to check and publish scheduled posts
- **Notification System:** Sends notifications to blog followers when posts are published
- **Search Filters:** Filter posts by Published/Not Published status and group by Publication Date or Scheduled Publication Date
### Technical Details
- Extends blog.post model with new fields (scheduled_publication_date, schedule_type, next_departure) and methods
- Includes a transient model blog.post.schedule.date for the scheduling wizard
- Inherits and enhances form and search views from website_blog
- Handles edge cases: setting past dates, clearing scheduled dates, multiple record writes
- Comprehensive test coverage with 25+ test cases
### Dependencies
- `website_blog`
### Test Plan
- [x] Install the module on a clean database
- [x]  Create a blog post with a future scheduled date and verify it's not published immediately
- [x]  Create a blog post with a past scheduled date and verify it publishes immediately
- [x]  Test the Schedule wizard dialog
- [x]  Test the Publish Now button
- [x]  Run the cron job manually and verify scheduled posts are published
- [x]  Verify notifications are sent to blog followers
- [x]  Test search filters and grouping options
- [x]  Run all unit tests: pytest website_blog_scheduled_publication/tests/

cc @ivs-cetmix @StefanRijnhart @marcelsavegnago @kaynnan @WesleyOliveira98

### CODE REVIEW

**Review by kaynnan — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Review by WesleyOliveira98 — APPROVED** [review]
LGTM

**Comment by kaynnan** [comment]
ping @OCA/website-maintainers

File: website_blog_scheduled_publication/models/blog_post.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    @api.model
+    def _publish_scheduled_posts(self):
+        """Cron job method to publish scheduled blog posts."""
+        now = fields.Datetime.now()
+        scheduled_posts = self.search(
+            [
+                ("scheduled_publication_date", "<=", now),
+                ("scheduled_publication_date", "!=", False),
+                ("website_published", "=", False),
+                ("active", "=", True),
```
Comment (tarteo) [inline website_blog_scheduled_publication/models/blog_post.py:31]: nitpicking: active domain leaf is not needed here as these are filtered by default.

**Review by tarteo — APPROVED** [review]
